Update partial reindexing logsearch docs

ccostino opened the pull request:
This changeset updates our instructions for a partial day reindex with logsearch based on our most recent quarterly backup exercise.

## Changes proposed in this pull request:
- A note about pausing pipelines in Concourse before starting
- Reorganizing of steps to make it easier to follow
- Expanded information on providing specific timeframes
- Added instructions for modifying an IAM role policy
- Correction for the rm statement to cleanup previous reindexing attempts
- Improved documentation for monitoring the progress of reindexing

:sunglasses: [PREVIEW URL](https://cg-88d42ca6-59d7-47e0-9500-4dd9251360b9.app.cloud.gov/preview/cloud-gov/cg-site/update-partial-log-restore-docs/docs/ops/runbook/troubleshooting-logsearch/#reindexing-data-from-s3-partial-day)

## Security considerations
- Some mention of IAM role policy adjustments, however these instructions are pulled from previous work done publicly in a [cg-provision PR](https://github.com/cloud-gov/cg-provision/pull/798)

ccostino committed:
Update partial reindexing logsearch docs

This changeset updates our instructions for a partial day reindex with logsearch based on our most recent quarterly backup exercise.  This includes the following changes:
- A note about pausing pipelines in Concourse before starting
- Reorganizing of steps to make it easier to follow
- Expanded information on providing specific timeframes
- Added instructions for modifying an IAM role policy
- Correction for the rm statement to cleanup previous reindexing attempts
- Improved documentation for monitoring the progress of reindexing

bengerman13 commented on _docs/ops/runbook/troubleshooting-logsearch.md:
````diff
@@ -123,35 +130,101 @@ mutate {
 index => "logs-%{@index_type}-%{[@metadata][index-date]}"
 ```
 
+### Disable the timecop filter
+
+The default Logsearch-for-CloudFoundry configuration will drop any messages older than 24 hours. When reindexing older data this sanity check needs to be disabled.
+
+To disable the timecop filter set the environment variable `TIMECOP_REJECT_LESS_THAN_HOURS` to match the desired rentention policy:
+
+```sh
+export TIMECOP_REJECT_LESS_THAN_HOURS=$((180 * 24))
+```
+
````
this is already on this page, isn't it?

ccostino replied:
Yes, it is probably being flagged as new because I moved it, but I will double check!

bengerman13 commented on _docs/ops/runbook/troubleshooting-logsearch.md:
````diff
@@ -123,35 +130,101 @@ mutate {
 ```sh
 export JAVA_HOME=/var/vcap/packages/openjdk-11/jre
-rm /tmp/s3_import.sincedb
+rm /var/vcap/data/ingestor_syslog/tmp/s3_import.sincedb
````
this is only necessary if you iterate on this multiple times, so we should remove it

ccostino replied:
I'm adding a new line to call this out just in case, but removing it from the steps here.

bengerman13 commented on _docs/ops/runbook/troubleshooting-logsearch.md:
````diff
@@ -123,35 +130,101 @@ mutate {
+Logstash will run forever once started. Monitor the progress of the reindex, and stop logstash once the data has been reindexed. Progress can be monitored by first setting turning the logstash reindexing to a background process and then by tailing the sincedb file which logstash will update after each file it processes:
 
 ```sh
+# While logstash process is running in your foreground:
+<Ctrl>+Z
+bg
+disown
+
````
I don't think this is necessary to document - there are several ways to accomplish this that any terminal user should be able to figure out on their own

ccostino replied:
That's true, but in the heat of the moment when dealing with an incident, the last thing I want to do is have to look up things that escape my brain; much easier to have the steps laid out right there!

bengerman13 commented on _docs/ops/runbook/troubleshooting-logsearch.md:
````diff
@@ -123,35 +130,101 @@ mutate {
+```
+
+Now save the policy, and you're done!
+
````
I think we should just reference terraform here, in case these permissions change in the future

ccostino replied:
Done!

apburnes commented on _docs/ops/runbook/troubleshooting-logsearch.md:
```diff
@@ -67,7 +70,8 @@ cp /var/vcap/jobs/ingestor_syslog/config/logstash.conf /var/vcap/data/ingestor_s
 Edit the `/tmp/logstash-restore.conf` and make the following changes:
 
-#### Update the "input" block to only contain s3
+#### Update the "input" block to only contain S3 information
```
If you want to run multiple days, hours or minutes, you can declare multiple s3 blocks with the unique prefix paths (ie `2021/01/00, 2021/01/01, 2021/01/02/00`) to run multiple batches of data without needing to rerun and update the s3 block settings.  Each block just needs a unique prefix and unique `.sincedb` file

bengerman13 replied:
oh, yeah, this was wonderful once we figured it out!

ccostino replied:
Will add it in, thank you!

ccostino replied:
Done, see below!

ccostino committed:
Made some adjustments based on feedback:

- Added steps for ingesting multiple time periods at once
- Removed unecessary steps
- Improved instructions on IAM role policy adjustments

h/t to @bengerman13 and @apburnes for the feedback and suggestions!
